packages/nimble-components/src/table/specs/table-column-specs/table-column-file-size.md

```diff
@@ -0,0 +1,222 @@
+# File Size Table Column
+
+## Overview
+
+The file size table column will provide a way to visualize numeric data that represents a file size (or other binary data size). It will automatically convert the raw value to use an appropriate display unit.
+
+### Background
+
+[Work Item](https://ni.visualstudio.com/DevCentral/_workitems/edit/2526114)
+
+### Non-goals
+
+-   Configurable number of decimal places
+-   Configurable units (e.g. display all values in KB)
+-   Grouping by value ranges/buckets
+
+### Features
+
+-   Converts the byte value to largest unit (e.g. KB, MB, GB) that results in a value of >=1
+-   Option to use 1024-based units (e.g. KiB) or 1000-based units (e.g. KB)
+-   Displays up to one decimal place
+-   Sorts/groups by numeric value (number of bytes) rather than display string
+-   Localized number and unit strings
+
+### Risks and Challenges
+
+### Prior Art/Examples
+
+Size column in [SLE Files grid](https://dev.lifecyclesolutions.ni.com/files/)
+
+![SLE Files grid](../spec-images/fileSizesSle.png)
+
+Windows explorer (details view)
+
+![Windows explorer](../spec-images/fileSizesWindows.png)
+
+Chrome Dev Tools Network tab
+
+![Chrome Dev Tools Network tab](../spec-images/fileSizeChromeDevTools.png)
+
+---
+
+## Design
+
+### API
+
+`nimble-table-column-file-size`
```

Comment:
For that specific example, I highly doubt we would have a progress-bar-like column that is _only_ meant to show progress of files/file sizes. I'm having a hard time coming up with another presentation (for file/memory sizes, specifically) that is plausible. I'm not putting my foot down, but I think this may be a case where we don't need to tack `-text` onto the end of the element name.

Reply:
> progress-bar-like column

 I could imagine that. 
![image](https://github.com/ni/nimble/assets/1588923/3198b238-a1b7-4038-959e-03d327a90016)

And a text representation of progress
![image](https://github.com/ni/nimble/assets/1588923/2529528e-c711-4947-94c0-6d48620828d0)

And both from the example given below:
![image](https://github.com/ni/nimble/assets/1588923/4265e562-3fdd-494d-9197-bce2cf1249a2)

Particularly for things like systems grid representing software being installed to devices with limited space. Could imagine that being a two segment progress bar where current used space is filled, how much this package adds, and total space is the whole bar. Or one segment where the whole bar is just the available space and segment is how much the package will use.

> If so maybe we should update those guidelines to describe that case more generally.

I think what that case is describing is that at the time it was expected that the same column might have multiple visualizations (text or icon+text). Instead of having separate columns for those visualizations. I dunno if that was a great choice and if we would actually implement that. It would probably be a new column element depending on the requirement (current requirements sound like they want custom messages per mapping, ie a warning status with a row specific message, so that would be a pretty different API and likely a different column). I think that was a bad degree of flexibility to leave in because I think it's very likely that different visualizations have different visualization specific api configuration that would be conflated together if configured on a single column.

Some examples based on this (explicit visuals):

- `nimble-table-column-file-size-text`
- `nimble-table-column-file-size-progress`
- `nimble-table-column-file-size-progress-text`

vs not explicit visuals:

- `nimble-table-column-file-size` (renders as text, but not clear)
- `nimble-table-column-file-size-progress` (renders as progress bar or text but not clear)

I think I lean towards the explicit set of:

- `nimble-table-column-file-size-text` (variant file-size)
- `nimble-table-column-file-size-progress` (variant file-size)
- `nimble-table-column-file-size-progress-text` (variant file-size and variant progress)
- `nimble-table-column-number-text` (variant number)
- `nimble-table-column-progress` (implicit variant number)
- `nimble-table-column-progress-text` (implicit variant number, variant progress)

With the thoughts:
The progress* columns probably have similar apis for configuring the data set for file size progress (configuring the row size and total size) distinct from text
The progress visual column probably has a distinct api for configuring the visual from text

And I might suggest the opposite for a docs clarification. Shoving or anticipating multiple visuals for the same column is probably not the right direction.

So I think a naming update could be:

- Always have the visual in the column name (mapping is out of compliance with the updated rule and should be mapping-text, but may be a low ROI to update or this might be the **best** time to fix it since most usage of nimble-table is under our control in sl-lib 🤷‍♂️)
- Add the concept of "implicit variants" where variants can be implicit if they are simple / primitive types / intuitive defaults for a given visual

Reply:
Devtools research on trying to be consistent: https://goo.gle/devtools-si

Reply:
Ok, I will change to `nimble-table-column-file-size-text` and look at updating the naming guidelines.

> mapping is out of compliance with the updated rule and should be mapping-text

We don't have a column named "mapping" -- it's `nimble-table-column-enum-text`, so I think we're compliant.

Reply:
Now renaming to `nimble-table-column-byte-text`.